File: java-algorithm/src/main/java/com/leetcode/Problem881.java
package com.leetcode;

import java.util.Arrays;

/**
 * 881. Boats to Save People
 * https://leetcode.com/problems/boats-to-save-people/
 */
public class Problem881 {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int i = 0, j = people.length - 1;
        int ans = 0;
        while (i < j) {
            if (people[i] + people[j] <= limit) i++;
            ans += 1;
            j--;
        }
        if (i == j) {
            ans++;
        }
        return ans;
    }
}

File: python-algo/leetcode/problem_881.py
"""881. Boats to Save People
https://leetcode.com/problems/boats-to-save-people/
"""
from typing import List


class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        ans = 0
        i, j = 0, len(people) - 1
        while i < j:
            if people[i] + people[j] <= limit:
                i += 1
            j -= 1
            ans += 1
        if i == j:
            ans += 1
        return ans

